lib/wallet/walletdb.js

```diff
@@ -206,6 +206,65 @@ class WalletDB extends EventEmitter {
       wallet.id, wallet.wid, addr.toString(this.network));
 
     this.primary = wallet;
+
+    await this.migrateChange();
+  }
+
+  /**
+   * Run change address migration.
+   * @param {Wallet} wallet
+   */
+
+  async migrateChange() {
+    if (this.options.memory)
+      return;
+
+    if (await this.db.has(layout.M.encode(0)))
+      return;
+
+    const b = this.db.batch();
+
+    b.put(layout.M.encode(0), null);
+
+    this.logger.warning('Checking change address corruption...');
+
+    const wids = await this.db.keys({
+      gte: layout.W.min(),
+      lte: layout.W.max(),
+      parse: key => layout.W.decode(key)[0]
+    });
+
+    let total = 0;
+
+    for (const wid of wids) {
+      const wallet = await this.get(wid);
+
+      this.logger.info('Checking wallet (id=%s, wid=%d).',
+                       wallet.id, wid);
+
+      total += await wallet.migrateChange(b);
+    }
+
+    if (this.options.migrate === 0 || total === 0)
+      await b.write();
+
```

Comment:
Just reasoning about the control flow here. There is no need to rescan if the total migrated is 0, so skip the inner if/else block. Only rescan if the user specifically opts in, since it is a more costly operation.

Reply:
This will write the migration ID, though.

Reply:
cACK - this will be easy for people running infra in the cloud, compared to running a script, they just will need to update their config file. Go for spelling it out in the changelog